org.opent2t.sample.thermostat.superpopular/com.nest.thermostat/js/thingTranslator.js

```diff
@@ -275,6 +278,133 @@ function validateResourceGet(resourceId) {
     }
 }
 
+/**
+ * If the user provides a unit, validate the value is within the units range.
+ *  Range is defined by Nest as:
+ *    c [9-32]
+ *    f [50-90]
+ *  Unless, is_locked = true, then it will be defined by the locked values.
+ * 
+ * If the unit is provided and within range, returns provided unit 
+ *  otherwise an exception is thrown.
+ * 
+ * If no unit is provided and the value is within one of the two ranges,
+ *  the unit of the valid range will be returned, otherwise an 
+ *  exception is thrown.
+ * 
+ * TODO: Could use the 'temperature_scale' from the thermostat to pick a unit.
+ * 
+ * @param {*} resourceSchema 
+ * @param {*} providerSchema 
+ */
+function getValidatedUnit(resourceSchema, providerSchema) {
+
+    var value = resourceSchema.temperature;
+
+    if (resourceSchema.hasOwnProperty('units') && resourceSchema.units != null) {
```

Comment:
Keep in mind that undefined is also a valid value in JS... it's possible to define a property 'units' and assign it undefined:

resourceSchema.units = undefined

...or:

resourceSchema.units = <<some computation that produces the undefined value>>

..which would thgen cause this type of an if condition to misbehave.

Reply:
Agreed. Just do resourceSchema.units === undefined || resourceSchema.units === null

Or better define a helper method.
isNullOrUndefined

Reply:
Fixed via helper method.